src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -94,9 +94,17 @@ protected virtual string GetResourcePrefix(TypeInfo typeInfo, string baseNamespa
                 throw new ArgumentNullException(nameof(baseNamespace));
             }
 
-            return string.IsNullOrEmpty(resourcesRelativePath)
-                ? typeInfo.FullName
-                : baseNamespace + "." + resourcesRelativePath + TrimPrefix(typeInfo.FullName, baseNamespace + ".");
+            if (string.IsNullOrEmpty(resourcesRelativePath))
+            {
+                return typeInfo.FullName;
```

Comment:
Should this consider the `baseNamespace` here as one could have a RootNamespaceAttribute but no resources relative path?

Reply:
To my knowledge if you don't have a `resourcesRelativePath` the the resource name is just the full type name.

Reply:
Why not write a test for this case? or you can find out from a test app in VS

Reply:
`Create_ResourceLocationAttribute_RootNamespaceIgnoredWhenNoLocation` has an odd name but is that test.